### ISSUE TITLE
Query params values in JSON output have NUL replaced by .

### ISSUE THREAD

**Issue by emanuele6** [curl/trurl#205]
NUL is replaced by `.` also in the JSON output, not just in the output of `-g '{query:}'` and `-g '{query-all:}'`.

Since `JSON` supports NUL in strings, and values in params are URL decoded, I think it should not be replaced.
```diff
 [
   {
     "url": "http://example.org/?q=mr%00smith",
     "parts": {
       "scheme": "http",
       "host": "example.org",
       "path": "/",
       "query": "q=mr%00smith"
     },
     "params": [
       {
         "key": "q",
-        "value": "mr.smith"
+        "value": "mr\u0000smith"
       }
     ]
   }
 ]
```

**Comment by bagder** [comment]
Fair point!

**Comment by bagder** [comment]
When we fix this, we need to change `sortquery()` if we decide to keep the nulls embedded in the strings, as then `strcasecmp` will not suffice.

`qpairs[]` and `qpairsdec[]` probably need to be converted to a `struct string` or something that keeps a pointer and a length separate.

**Comment by bagder** [comment]
Fixed by #214
